<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>KeyGenDialog</class>
 <widget class="QDialog" name="KeyGenDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>475</width>
    <height>635</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Key Generation</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <property name="leftMargin">
    <number>5</number>
   </property>
   <property name="topMargin">
    <number>5</number>
   </property>
   <property name="rightMargin">
    <number>5</number>
   </property>
   <property name="bottomMargin">
    <number>5</number>
   </property>
   <item>
    <layout class="QFormLayout" name="formLayout_2">
     <property name="fieldGrowthPolicy">
      <enum>QFormLayout::FieldGrowthPolicy::ExpandingFieldsGrow</enum>
     </property>
     <property name="rowWrapPolicy">
      <enum>QFormLayout::RowWrapPolicy::DontWrapRows</enum>
     </property>
     <property name="labelAlignment">
      <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
     </property>
     <property name="formAlignment">
      <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
     </property>
     <item row="1" column="0">
      <widget class="QLabel" name="nameLabel">
       <property name="text">
        <string>Name</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLineEdit" name="nameEdit"/>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="emailLabel">
       <property name="text">
        <string>Email</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLineEdit" name="emailEdit"/>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="commentLabel">
       <property name="text">
        <string>Comment</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLineEdit" name="commentEdit"/>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="keyDBLabel">
       <property name="text">
        <string>Key Database</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QComboBox" name="keyDBIndexComboBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QTabWidget" name="tabWidget">
       <property name="tabPosition">
        <enum>QTabWidget::TabPosition::North</enum>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <property name="documentMode">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="easyTab">
        <attribute name="title">
         <string>Easy Mode</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <property name="bottomMargin">
          <number>5</number>
         </property>
         <item>
          <layout class="QGridLayout" name="gridLayout">
           <item row="5" column="0">
            <widget class="QLabel" name="combinationLabel">
             <property name="text">
              <string>Combination</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="easyProfileLabel">
             <property name="text">
              <string>Profile</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1" colspan="2">
            <widget class="QComboBox" name="easyValidityPeriodComboBox"/>
           </item>
           <item row="5" column="1" colspan="2">
            <widget class="QComboBox" name="easyCombinationComboBox"/>
           </item>
           <item row="0" column="1" colspan="2">
            <widget class="QComboBox" name="easyProfileComboBox"/>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="easyValidPeriodLabel">
             <property name="text">
              <string>Validity Period</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Orientation::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QPushButton" name="deletePushButton">
             <property name="text">
              <string>Delete Profile</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="savePushButton">
             <property name="text">
              <string>Save Profile</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="pTab">
        <attribute name="title">
         <string>Primary Key</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="2" column="0">
            <widget class="QLabel" name="pExpireDateLabel">
             <property name="text">
              <string>Expire Date</string>
             </property>
            </widget>
           </item>
           <item row="7" column="0" colspan="3">
            <widget class="QCheckBox" name="pExpireCheckBox">
             <property name="text">
              <string>Non Expired</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1" colspan="2">
            <widget class="QDateTimeEdit" name="pExpireDateTimeEdit"/>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="pAlgoLabel">
             <property name="text">
              <string>Algorithm</string>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="pUsageLabel">
             <property name="text">
              <string>Usage</string>
             </property>
            </widget>
           </item>
           <item row="5" column="1" colspan="2">
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QCheckBox" name="pEncrCheckBox">
               <property name="text">
                <string>Encrypt</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="pSignCheckBox">
               <property name="text">
                <string>Sign</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="pAuthCheckBox">
               <property name="text">
                <string>Authentication</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="0" column="1" colspan="2">
            <widget class="QComboBox" name="pAlgoComboBox"/>
           </item>
           <item row="6" column="0" colspan="3">
            <widget class="QCheckBox" name="noPassphraseCheckBox">
             <property name="text">
              <string>No Passphrase</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="pKeyLengthLabel">
             <property name="text">
              <string>Key Length</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1" colspan="2">
            <widget class="QComboBox" name="pKeyLengthComboBox"/>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Orientation::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="sTab">
        <attribute name="title">
         <string>Subkey</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_7">
         <item>
          <layout class="QGridLayout" name="gridLayout_4">
           <item row="2" column="0">
            <widget class="QLabel" name="sExpireDateLabel">
             <property name="text">
              <string>Expire Date</string>
             </property>
            </widget>
           </item>
           <item row="6" column="0" colspan="3">
            <widget class="QCheckBox" name="sExpireCheckBox">
             <property name="text">
              <string>Non Expired</string>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="sUsageLabel">
             <property name="text">
              <string>Usage</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1" colspan="2">
            <widget class="QDateTimeEdit" name="sExpireDateTimeEdit"/>
           </item>
           <item row="0" column="1" colspan="2">
            <widget class="QComboBox" name="sAlgoComboBox"/>
           </item>
           <item row="5" column="1" colspan="2">
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QCheckBox" name="sEncrCheckBox">
               <property name="text">
                <string>Encrypt</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="sSignCheckBox">
               <property name="text">
                <string>Sign</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="sAuthCheckBox">
               <property name="text">
                <string>Authentication</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="sAlgoLabel">
             <property name="text">
              <string>Algorithm</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="sKeyLengthLabel">
             <property name="text">
              <string>Key Length</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1" colspan="2">
            <widget class="QComboBox" name="sKeyLengthComboBox"/>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer_3">
           <property name="orientation">
            <enum>Qt::Orientation::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPlainTextEdit" name="statusPlainTextEdit">
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="generateButton">
     <property name="text">
      <string>Generate</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>easyProfileComboBox</tabstop>
  <tabstop>easyValidityPeriodComboBox</tabstop>
  <tabstop>easyCombinationComboBox</tabstop>
  <tabstop>statusPlainTextEdit</tabstop>
  <tabstop>generateButton</tabstop>
  <tabstop>pExpireCheckBox</tabstop>
  <tabstop>pExpireDateTimeEdit</tabstop>
  <tabstop>pEncrCheckBox</tabstop>
  <tabstop>pSignCheckBox</tabstop>
  <tabstop>pAuthCheckBox</tabstop>
  <tabstop>pAlgoComboBox</tabstop>
  <tabstop>noPassphraseCheckBox</tabstop>
  <tabstop>pKeyLengthComboBox</tabstop>
  <tabstop>sExpireCheckBox</tabstop>
  <tabstop>sExpireDateTimeEdit</tabstop>
  <tabstop>sAlgoComboBox</tabstop>
  <tabstop>sEncrCheckBox</tabstop>
  <tabstop>sSignCheckBox</tabstop>
  <tabstop>sAuthCheckBox</tabstop>
  <tabstop>sKeyLengthComboBox</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
